src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseController.java

```diff
@@ -100,6 +101,29 @@ public CaseData updateCase(
         }
     }
 
+    @PostMapping(value = "{caseId}/{eventId}/case-update", consumes = APPLICATION_JSON, produces = APPLICATION_JSON)
+    @Operation(description = "Updating casedata")
+    public CaseData caseUpdate(
+        @NotNull @Valid @RequestBody UpdateCaseData updateCaseData,
+        @PathVariable("eventId") String eventId,
+        @PathVariable("caseId") String caseId,
+        @RequestHeader(HttpHeaders.AUTHORIZATION) @Parameter(hidden = true) String authorisation,
+        @RequestHeader(PrlAppsConstants.SERVICE_AUTHORIZATION_HEADER) String s2sToken
+    ) throws JsonProcessingException {
+        if (isAuthorized(authorisation, s2sToken)) {
+            CaseDetails caseDetails = null;
+            caseDetails = caseService.updateCaseDetails(
+                authorisation,
+                caseId,
+                eventId,
```

Comment:
Do we realy need eventId? if yes , what are the possible values?

Reply:
Yes shashi.. we require this because the internal function call stack which we are using is requiring eventId to be passed ... i.e : **caseRepository.updateCase(authToken, caseId, caseData, CaseEvent.fromValue(eventId));** . And the possible values are  which we have in caseEvent.java and also In ccd .. caseEvent.json , authorizations should also be given for them.